core/tests/transport_upgrade.rs

```diff
@@ -82,7 +82,7 @@ fn upgrade_pipeline() {
     let listener_keys = identity::Keypair::generate_ed25519();
     let listener_id = listener_keys.public().to_peer_id();
     let mut listener_transport = MemoryTransport::default()
-        .upgrade(upgrade::Version::V1)
+        .upgrade()
```

Comment:
Technically we could remove `upgrade` now, right?

What do you think of deprecating `upgrade` first and instead offer an `authenticate` function for each transport that simply implies `upgrade(V1Lazy)`?

Reply:
At the moment it is a breaking change. Deprecating it first and offering a new API that starts with `authenticate` straight from `Transport` is not.